apps/token/src/routes/governance/propose/raw/propose-raw.tsx

```diff
@@ -0,0 +1,107 @@
+import { useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { useEnvironment } from '@vegaprotocol/environment';
+import { Heading } from '../../../../components/heading';
+import { VegaWalletContainer } from '../../../../components/vega-wallet-container';
+import {
+  FormGroup,
+  InputError,
+  Link,
+  TextArea,
+} from '@vegaprotocol/ui-toolkit';
+import { useProposalSubmit } from '@vegaprotocol/governance';
+import {
+  ProposalFormSubmit,
+  ProposalFormTransactionDialog,
+} from '../../components/propose';
+
+export interface RawProposalFormFields {
+  rawProposalData: string;
+}
+
+export const ProposeRaw = () => {
+  const { VEGA_EXPLORER_URL, VEGA_DOCS_URL } = useEnvironment();
+  const { t } = useTranslation();
+  const {
+    register,
+    handleSubmit,
+    formState: { isSubmitting, errors },
+  } = useForm<RawProposalFormFields>();
+  const { finalizedProposal, submit, Dialog } = useProposalSubmit();
+
+  const hasError = Boolean(errors.rawProposalData?.message);
+
+  const onSubmit = async (fields: RawProposalFormFields) => {
+    await submit(JSON.parse(fields.rawProposalData));
+  };
+
+  return (
+    <>
+      <Heading title={t('NewRawProposal')} />
+      <VegaWalletContainer>
+        {() => (
+          <>
+            {VEGA_DOCS_URL && (
+              <p className="text-sm">
+                <span className="mr-1">{t('ProposalTermsText')}</span>
+                <Link
+                  href={`${VEGA_DOCS_URL}/tutorials/proposals`}
+                  target="_blank"
+                >{`${VEGA_DOCS_URL}/tutorials/proposals`}</Link>
+              </p>
+            )}
```

Comment:
chore (non-blockin): This is repeated a lot, might be worth making it a component